app/lib/screens/market/buy_tft.dart

```diff
@@ -130,22 +129,50 @@ class _BuyTFTWidgetState extends State<BuyTFTWidget> {
       });
       return false;
     }
-    final balance = roundAmount(availableUSDC ?? '0');
 
-    if (balance - Decimal.parse(amount) <= Decimal.zero) {
+    try {
+      final decimalAmount = Decimal.parse(amount);
+
+      if (decimalAmount < Decimal.zero) {
+        setState(() {
+          amountError = 'Amount cannot be negative';
+        });
+        return false;
+      }
+
+      if (decimalAmount == Decimal.zero) {
+        setState(() {
+          amountError = 'Amount must be greater than zero';
+        });
+        return false;
+      }
+
+      final balance = roundAmount(availableUSDC ?? '0');
+
+      if (balance - decimalAmount <= Decimal.zero) {
+        setState(() {
+          amountError = 'Balance is not enough';
+        });
+        return false;
+      }
+
+      if (decimalAmount > Decimal.parse(availableUSDC ?? '0')) {
+        setState(() {
+          amountError = 'Not enough balance';
+        });
+        return false;
```

Comment:
What is the difference between these 2 checks?

Reply:
- The first one for negative numbers.
- second one for checking that amount not exceeding users' balance of USDC